### ISSUE TITLE
Reverting to old version when deploying

### ISSUE THREAD

**Issue by karlvd** [ionic-team/cordova-plugin-ionic#23]
I have cordova-plugin-ionic installed using background mode
`ionic cordova plugin add cordova-plugin-ionic --save --variable APP_ID="xxxxxx" --variable CHANNEL_NAME="Master" --variable UPDATE_METHOD="background"
`
Everytime I deploy a new version to Ionic Pro the following will happen:
1. Deploying a new version to Ionic Pro via Git to the Master Channel
2. Shutting down app to trigger background update on next restart
3.  The next version deployed to the phone is the version deployed originally via Xcode
4. Shutting down app to trigger background update again
5. The correct version is deployed to the app (The one marked Master in Ionic Pro)

I have tried redeploying via XCode, reinstalling plugin etc. but looks like it´s some bug, IMO the latest version should be downloaded and updated on restart.

Testing in iOS device, iphone 7

```
Ionic Info
cli packages: (C:\Users\kado\AppData\Roaming\npm\node_modules)

    @ionic/cli-utils  : 1.17.0
    ionic (Ionic CLI) : 3.17.0

global packages:

    cordova (Cordova CLI) : not installed

local packages:

    @ionic/app-scripts : 3.0.1
    Cordova Platforms  : android 6.1.2
    Ionic Framework    : ionic-angular 3.8.0

System:

    Node : v6.11.2
    npm  : 3.10.10
    OS   : Windows 10

Environment Variables:

    ANDROID_HOME : not set

Misc:

    backend : pro
```

**Comment by joeldavuk** [comment]
I have experienced this behaviour too, in manual mode as well. If you do use different channels check the branch config.xml and package.json has the same channel you want to deploy to and the build is from that channel. That seemed to solve it for me.

Needs to be more documentation on exactly how/if Ionic uses this information for the automatic updates.

**Comment by PauCodina** [comment]
I have the same issue in auto or manual mode.

After a new deploy, the first time that you shutts down the app, the deploy_uuid variable is set to NO_DEPLOY_LABEL instead the new uuid value. For theese reason, the app comes back to the version deployed originally via Xcode.

If you shutts down again the app, deploy_uuid is set correctly.

A possible solution would be that Pro.deploy.redirect() accepts a param with the desired uuid version (previously donwloaded).

Any idea?
Regards

**Comment by timbru31** [comment]
We are experiencing the same issue. 👎

**Comment by connor-odoherty** [comment]
This issue is still happening, can we get an update?

**Comment by Fuiste** [comment]
This was an issue related to the automatic redirect on app load.  We've just published plugin 4.1.7, which solved the issue on our devices.
